src/wallet.cpp

```diff
@@ -1283,7 +1291,7 @@ bool CWallet::CreateTransaction(const vector<pair<CScript, int64_t> >& vecSend,
                     nFeeRet += nMoveToFee;
                 }
 
-                if (nChange > 0)
+                if (nChange > 0 && !isData)
```

Comment:
The `&& !isData` check here would disable getting change sent back to the wallet when using `CreateTransaction` to send a message.